src-tauri/src/read_files.rs

```diff
@@ -0,0 +1,186 @@
+use serde_json::{json, Map, Value};
+use std::borrow::BorrowMut;
+use std::collections::HashMap;
+use std::fs;
+use std::io::{BufWriter, Write};
+use std::path::{Path, PathBuf};
+use xml2json_rs::JsonConfig;
+
+// This implementation is compatible with v4 of the JavaScript implementation
+const CACHE_VERSION: u8 = 4;
```

Comment:
I wonder if we should pull this variable out into package.json, so both rust and js can read it from there without duplication.

Reply:
Right now, the two implementations are kind of diverged since there's a bunch of fix-up being done in the rust `index` implementation to make things match up.

However, you are right that there is a failure mode right now were the JS side gets updated to v5, native code loads v4 and just passes it off and nothing else verifies that the version number is correct on the JS side.
- [ ] Double check cache version after we parse the cache in JS